packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,295 @@
+            const sometimes = () => {
+                attempt += 1;
+                return attempt !== 5 ? [1, 2, 3] : [];
+            };
+
+            await expect(sometimes).retry({ retries: 10 }).to.be.empty;
+
+            expect(attempt).to.equal(5);
+        });
+
+        it('.sealed', async () => {
+            const sealedObject = Object.seal({ prop1: 'value1', prop2: 'value2' });
+
+            await expect(() => sealedObject).retry().to.be.sealed;
+        });
+
+        it('.not.sealed', async () => {
+            const notSealedObject = { prop1: 'value1', prop2: 'value2' };
+
+            try {
+                await expect(() => notSealedObject).retry({ retries: 3 }).to.be.sealed;
+            } catch (error: unknown) {
+                expect(error).to.be.an.instanceOf(Error);
+                expect((error as Error).message).to.include('3 retries');
+            }
+        });
+    });
+
+    describe('should work with various assertion methods:', () => {
+        it('.satisfy()', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+                return { attempts };
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 5, delay: 10 })
+                .to.satisfy((obj: { attempts: number }) => obj.attempts > 3);
+
+            expect(attempts).to.equal(4);
+        });
+
+        it('.have.property(), and.be.above()', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+                return attempts < 4 ? { notValue: 2 } : { value: attempts };
+            };
+
+            await expect(funcToRetry).retry({ retries: 5, delay: 10 }).have.property('value').and.be.above(4);
+
+            expect(attempts).to.equal(5);
+        });
+
+        it('.to.increase(), .by()', async () => {
+            let attempts = 0;
+            const myObj = { val: 1 };
+
+            const addTwo = () => {
+                attempts++;
+                myObj.val += 2;
+            };
+
+            await expect(addTwo).retry().to.increase(myObj, 'val').by(2);
+
+            expect(attempts).to.equal(1);
+        });
+
+        it('.oneOf()', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+
+                return attempts;
+            };
+
+            await expect(funcToRetry).retry().to.not.be.oneOf([1, 2, 3]);
+            expect(attempts).to.equal(4);
+        });
+
+        it('.change()', async () => {
+            let attempts = 0;
+            const myObj = { dots: '', comas: '' };
+
+            const addDot = () => {
+                attempts++;
+                myObj.dots += '.';
+            };
+
+            await expect(addDot).retry().to.change(myObj, 'dots');
+            await expect(addDot).retry().to.not.change(myObj, 'comas');
+
+            expect(attempts).to.equal(2);
```

Comment:
what benefit checking for `attempts` gives us?

Reply:
@yurii-ve verifying that retrying operation finished after the right attempt